catroid/src/main/java/org/catrobat/catroid/formulaeditor/FormulaElement.java

```diff
@@ -270,10 +277,34 @@ public Object interpretRecursive(Sprite sprite) {
 					returnValue = 0d;
 					Log.e(getClass().getSimpleName(), Log.getStackTraceString(exception));
 				}
+				break;
+			case DISTANCE_TO_TOUCH_POSITION:
+				returnValue = sprite.look.getDistanceToTouchPositionInUserInterfaceDimensions();
+				break;
+			case DISTANCE_TO_SPRITE_POSITION:
+				returnValue = interpretDistanceToSpritePosition(sprite, value);
+				break;
 		}
 		return normalizeDegeneratedDoubleValues(returnValue);
 	}
 
+	private Object interpretDistanceToSpritePosition(Sprite currentSprite, String value) {
+		String secondSpriteName = value;
+		Sprite secondSprite;
+
+		try {
+			secondSprite = ProjectManager.getInstance().getSceneToPlay().getSpriteBySpriteName(secondSpriteName);
+		} catch (Resources.NotFoundException exception) {
+			return 0d;
+		}
```

Comment:
I should've tested it in Scratch and also do more research… returning `0` is fine (I guess). Many calculations in the FormulaElement return `0` if there is some kind of undefined intermediate result or state. The `InterpretationException` is just used if the **result** of the Formula is NaN.